typed-builder-macro/src/struct_info.rs

```diff
@@ -85,15 +87,27 @@ impl<'a> StructInfo<'a> {
                 empty_type()
             }
         }));
-        let init_fields_expr = self.included_fields().map(|f| {
-            f.builder_attr.via_mutators.as_ref().map_or_else(
-                || quote!(()),
-                |via_mutators| {
-                    let init = &via_mutators.init;
-                    quote!((#init,))
-                },
-            )
-        });
+        let builder_method_const = Rc::new(OnceCell::new());
```

Comment:
Respectfully WTF? Looking at the code, it seems like a simple:
```rust
let mut builder_method_const = true;
```
would suffice.

Reply:
Borrow checker was not happy about the moved value. It could be done with `Rc<Cell<bool>>` though.

Reply:
I tried it locally and it worked. Had to remove the `move` from the closure in the `map`, of course.

I speculate that it didn't work for you because you tried it before adding the `collect`. The compiler, in a way, was complaining about the very same bug that the `collect` solved.

Another option is to just do a second iteration:

```rust
let builder_method_const = self
    .included_fields()
    .filter_map(|field| field.builder_attr.via_mutators.as_ref())
    .all(|via_mutators| matches!(via_mutators.init, syn::Expr::Lit(_)))
    .then(|| quote!(const));
```

But of course - all of this is irrelevant because https://github.com/idanarye/rust-typed-builder/pull/123#discussion_r1359976876.